barretenberg/cpp/src/barretenberg/dsl/acir_proofs/acir_composer.cpp

```diff
@@ -91,17 +93,7 @@ bool AcirComposer::verify_proof(std::vector<uint8_t> const& proof, bool is_recur
     // Hack. Shouldn't need to do this. 2144 is size with no public inputs.
     builder_.public_inputs.resize((proof.size() - 2144) / 32);
 
-    // TODO: We could get rid of this, if we made the Noir program specify whether something should be
-    // TODO: created with the recursive setting or not. ie:
-    //
-    // #[recursive_friendly]
-    // fn main() {}
-    // would put in the ACIR that we want this to be recursion friendly with a flag maybe and the backend
-    // would set the is_recursive flag to be true.
-    // This would eliminate the need for nargo to have a --recursive flag
-    //
-    // End result is that we may just be able to get it off of builder_, like builder_.is_recursive_friendly
```

Comment:
I settled on this flag living on the composer as I didn't feel the need to thread it down to the builder. The builder is created when we a create a circuit anyway which is the same time this flag is set.